rust/arrow/src/datatypes/mod.rs

```diff
@@ -199,6 +207,81 @@ pub struct Field {
     metadata: Option<BTreeMap<String, String>>,
 }
 
+// Decimal (precision, scale) = Decimal(1, 2) = 1.00
+pub trait ArrowDecimalType: fmt::Debug + Send + Sync + FromStr + PartialEq {
+    const MAX_DIGITS: usize;
+
+    // fn into_json_value(self) -> Option<Value>;
+
+    fn get_byte_width_for_precision_scale(precision: usize, scale: usize) -> usize;
+
+    // Rescale scale part
+    fn rescale(&mut self, scale: usize);
+
+    // Try to parse string with precision, scale
+    fn parse(
+        string: &str,
+        precision: usize,
+        scale: usize,
+    ) -> std::result::Result<Self, ParseDecimalError>;
+
+    fn to_byte_slice(&self) -> Vec<u8>;
+
+    fn from_bytes_with_precision_scale(
+        bytes: &[u8],
+        precision: usize,
+        scale: usize,
+    ) -> Self;
+}
+
+#[derive(Debug)]
+pub enum DecimalType {
```

Comment:
@alamb @andygrove @Dandandan @nevi-me @jorgecarleitao

I created representations for Decimal as `Int32DecimalType`, `Int64DecimalType`, `Int128DecimalType` or `LargeDecimal`, which uses BigInt.

1. What is the best way in Rust to abstract representations of `DataType::Decimal` in Rust? Enum?
2. How should I work with `DecimalArray`?  Should make it as generic by passing representation type to it (but how should I detect type, match in everyplace?) or make it generic on top of DecimalType and get byte size from it?

Thanks

Reply:
Hi @ovr  -- sorry for the lack of response. I am not familiar with this code so I will need some non trivial time to review and think about your question. I hope to find some time this weekend to do so.